### DISCUSSION TITLE
PlugIn registration and sync issues. - Answered

### DISCUSSION THREAD

**Discussion by whitedl** [bmeares/Meerschaum#53] [General]
I installed a new plugin through register followed by sync --loop. Not quite sure what happened as only 14 records were added to my database even though the job that I created was running for 24 hours. Should have several hundred records. The plugin is just an updated version of a plugin running now for a couple weeks. I dropped the table and deleted the job and plugin registration. I expected to go through a new setup but it seems that is not occurring as I should have to enter my API Key at initialization but that step is skipped and when I run a basic sync now I get the attached error. Thx.
<img width="1643" alt="Screen Shot 2022-01-29 at 1 08 16 PM" src="https://user-images.githubusercontent.com/8012877/151672377-ec8cc583-1353-4223-96a6-6a32e6f19dbe.png">

**Comment by bmeares** [comment]
Hi @whitedl thanks for posting! The API key for the `twitter` plugin I had emailed is set in its `register()` function. You can remove this key with `mrsm delete config plugins` or open the file with `mrsm edit config plugins`.

You can inspect the output of your background jobs with `show logs` (or `show logs --nopretty`). In case the background jobs are giving you trouble, you can run your syncing command in another process manager like `crontab` or `tmux`. E.g. to run the command in crontab, you could do the following:

```
### Fetch new tweets once per minute
* * * * * python -m meerschaum sync pipes -c plugin:twitter >> ~/twitter.log
```

I ran a quick test with a more generic `query` parameter and am able to fetch new tweets with each sync (duplicates rows are filtered out by default). Check that you're able to fetch new rows and that the process is running without errors.

Thanks for asking, and I hope this helps!
